src/confluent_kafka/src/Admin.c

```diff
@@ -976,6 +976,537 @@ static PyObject *Admin_alter_configs (Handle *self, PyObject *args,
         return NULL;
 }
 
+/**
+ * @brief Convert py AclBinding to C
+ */
+static rd_kafka_AclBinding_t*
+Admin_py_to_c_AclBinding (const PyObject *py_obj_arg,
+                        char *errstr,
+                        size_t errstr_size) {
+        int restype, resource_pattern_type, operation, permission_type;
+        char *resname = NULL, *principal = NULL, *host = NULL;
+        rd_kafka_AclBinding_t* ret = NULL;
+
+        PyObject *py_obj = (PyObject *) py_obj_arg;
+        if(cfl_PyObject_GetInt(py_obj, "restype_int", &restype, 0, 0)
+            && cfl_PyObject_GetString(py_obj, "name", &resname, NULL, 0)
+            && cfl_PyObject_GetInt(py_obj, "resource_pattern_type_int", &resource_pattern_type, 0, 0)
+            && cfl_PyObject_GetString(py_obj, "principal", &principal, NULL, 0)
+            && cfl_PyObject_GetString(py_obj, "host", &host, NULL, 0)
+            && cfl_PyObject_GetInt(py_obj, "operation_int", &operation, 0, 0)
+            && cfl_PyObject_GetInt(py_obj, "permission_type_int", &permission_type, 0, 0)) {
+                    ret = rd_kafka_AclBinding_new(restype, resname, \
+                        resource_pattern_type, principal, host, \
+                        operation, permission_type, errstr, errstr_size);
+        }
+        if (resname) free(resname);
+        if (principal) free(principal);
+        if (host) free(host);
+        return ret;
+}
+
+/**
+ * @brief Convert py AclBindingFilter to C
+ */
+static rd_kafka_AclBindingFilter_t*
+Admin_py_to_c_AclBindingFilter (const PyObject *py_obj_arg,
+                        char *errstr,
+                        size_t errstr_size) {
+        int restype, resource_pattern_type, operation, permission_type;
+        char *resname = NULL, *principal = NULL, *host = NULL;
+        PyObject *py_name = NULL, *py_principal = NULL, *py_host = NULL;
+        PyObject *py_obj = (PyObject *) py_obj_arg;
+        rd_kafka_AclBindingFilter_t* ret = NULL;
+
+
+        cfl_PyObject_GetAttr(py_obj, "name", &py_name, NULL, 1);
+        cfl_PyObject_GetAttr(py_obj, "principal", &py_principal, NULL, 1);
+        cfl_PyObject_GetAttr(py_obj, "host", &py_host, NULL, 1);
+        int name_is_none = py_name == Py_None;
+        int principal_is_none = py_principal == Py_None;
+        int host_is_none = py_host == Py_None;
+
+        if(cfl_PyObject_GetInt(py_obj, "restype_int", &restype, 0, 1)
+            && (name_is_none || cfl_PyObject_GetString(py_obj, "name", &resname, NULL, 1))
```

Comment:
This isn't a hot path so it doesn't matter in practice, but it would be more efficient and make the code more readable by adding a "allow_None" parameter to cfl_PyObject_GetString() that makes it return the value as NULL with positive return value in case the attr is None.